packages/vite/src/node/server/index.ts

```diff
@@ -158,7 +158,7 @@ export interface ServerOptions extends CommonServerOptions {
 
 export interface ResolvedServerOptions extends ServerOptions {
```

Comment:
It looks like the reason we didn't caught this in the first place is because we are incorrectly merging the types here. I think this can be fixed with 
```suggestion
export interface ResolvedServerOptions extends Omit<ServerOptions, 'fs' | 'middlewareMode' | 'sourcemapIgnoreList'> {
```